### DISCUSSION TITLE
Query - not contains seems to have a limit of 124, can we do better or bug?

### DISCUSSION THREAD

**Discussion by Arnii** [ravendb/ravendb#12917] [General]
We have a query in our app that needs to exclude a lot of documents (by ID) from the query. We do this as follows:
`query.AndAlso().Not.ContainsAny(x => x.Id, exludedDocuments);` 
which works great, but if that list gets too large it seems to not work at all. 
_(and that excludedDocuments list comes from another document, so it's different everytime)_

I tried myself by querying in Raven Studio using `not id() in` and a whole lot of Id's, the limit seems to be 124 documents, more than 124 seems to ignore the "not contains" completely. I attached a video of me testing in Raven Studio.

https://user-images.githubusercontent.com/11922408/138888064-21bb3c1d-343f-4276-8988-647120c8e2c3.mp4

Are we doing something wrong/inefficient or could this be a bug?

**Comment by ppekrol** [comment]
This could be related to changes made here: https://issues.hibernatingrhinos.com/issue/RavenDB-17206
Or is another variation of that issue.

Could you send us a full repro maybe so we can investigate this faster?

**Reply by Arnii** [reply]
Sorry, the video is the best repro we have so far and currently do not have the time for further testing, but that issue you sent is exactly what we're dealing with! Sorry for the double post. I see it has already been fixed for a future version, so I'm happy!

**Reply by garayx** [reply]
you can try to install the nightly and check if your query works
https://hibernatingrhinos.com/builds?product=RavenDB&version=5.1&channel=Nightly&pageNr=1

**Comment by garayx** [comment]
Hi

Sorry I didn't see that the fix is already included in the RavenDB version that you are using, can you send us a [failing test](https://ravendb.net/docs/article-page/5.2/Csharp/start/test-driver#unittest) or sample project that reproduce the issue ?

**Comment by garayx** [comment]
Hi

We have a reproduce for this issue, and will fix it shorty
[https://issues.hibernatingrhinos.com/issue/RavenDB-17423](https://issues.hibernatingrhinos.com/issue/RavenDB-17423)

**Reply by Arnii** [reply]
Thank you for the repro. We didn't have the time yet to create a proper full repro

**Comment by arekpalinski** [comment]
PR: https://github.com/ravendb/ravendb/pull/12984
